src/tracers.jl

```diff
@@ -12,147 +12,142 @@ The provided index set type `S` has to satisfy the following conditions:
 Subtypes of `AbstractSet{<:Integer}` are a natural choice, like `BitSet` or `Set{UInt64}`.
 """
 
+empty_sparse_vector(T) = T()
+empty_sparse_matrix(T) = T()
+
+sparse_vector(T, index) = T([index])
+sparse_vector(::Type{T}, index) where {T<:DuplicateVector} = T(index)
+sparse_vector(::Type{T}, index) where {T<:SortedVector} = T(index)
+
+#================#
+# Elementwise OR #
+#================#
+
+## We use ∨ to represent the elementwise OR
+# TODO REVIEW: \vee looks too much like the character 'v' in code
+# Gradient representations
+∨(a::G, b::G) where {G<:AbstractSet} = a ∪ b
+∨(a::G, b::G) where {G<:DuplicateVector} = G(vcat(a.data, b.data))
+∨(a::G, b::G) where {G<:SortedVector} = a ∪ b
+∨(a::G, b::G) where {G<:RecursiveSet} = a ∪ b
+
+# Hessian representations
+# TODO REVIEW: for now, we only support representing Hessians as a set of tuples
```

Comment:
We can support `AbstractMatrix{Bool}` for free

Reply:
See discussion above.